lib/checkstl.cpp

```diff
@@ -843,6 +846,105 @@ static bool isInvalidMethod(const Token * tok)
     return false;
 }
 
+struct InvalidContainerAnalyzer {
+    struct Info {
+        struct Reference {
+            const Token* tok;
+            ErrorPath errorPath;
+        };
+        std::unordered_map<int, Reference> expressions;
+        ErrorPath errorPath;
+        void add(const std::vector<Reference>& refs)
+        {
+            for (const Reference r : refs) {
+                add(r);
+            }
+        }
+        void add(const Reference& r)
+        {
+            if (!r.tok)
+                return;
+            expressions.insert(std::make_pair(r.tok->exprId(), r));
+        }
+
+        std::vector<Reference> invalidTokens() const
+        {
+            std::vector<Reference> result;
+            std::transform(expressions.begin(), expressions.end(), std::back_inserter(result), SelectMapValues{});
+            return result;
+        }
+    };
+    std::unordered_map<const Function*, Info> invalidMethods;
+
+    std::vector<Info::Reference> invalidatesContainer(const Token* tok) const
+    {
+        std::vector<Info::Reference> result;
+        if (Token::Match(tok, "%name% (")) {
+            const Function* f = tok->function();
+            if (!f)
+                return result;
+            ErrorPathItem epi = std::make_pair(tok, "Calling function " + tok->str());
+            const bool dependsOnThis = exprDependsOnThis(tok->next());
+            auto it = invalidMethods.find(f);
+            if (it != invalidMethods.end()) {
+                std::vector<Info::Reference> refs = it->second.invalidTokens();
+                std::copy_if(refs.begin(), refs.end(), std::back_inserter(result), [&](const Info::Reference& r) {
+                    const Variable* var = r.tok->variable();
+                    if (!var)
+                        return false;
+                    if (dependsOnThis && !var->isLocal() && !var->isGlobal() && !var->isStatic())
+                        return true;
+                    if (!var->isArgument())
+                        return false;
+                    if (!var->isReference())
+                        return false;
+                    return true;
+                });
+                std::vector<const Token*> args = getArguments(tok);
+                for (Info::Reference& r : result) {
```

Comment:
Is result nonempty here?

Reply:
No.

Reply:
so I guess you will fix something here before we merge it?

Reply:
What needs to be fixed?

Reply:
I wonder .. if `result` is always empty. Is the loop redundant?

Reply:
Its not always empty. The `copy_if` fills the `result`.

Reply:
ok thanks for that clarification I somehow missed it.